pkgs/native_assets_cli/lib/src/api/build_config.dart

```diff
@@ -32,168 +46,173 @@ abstract class BuildConfig {
   /// dependency of another. For this it is convenient to know the packageRoot.
   Uri get packageRoot;
 
-  /// The target being compiled for.
-  ///
-  /// Not available in [dryRun].
-  Target get target;
-
   /// The architecture being compiled for.
   ///
-  /// Not available in [dryRun].
-  Architecture get targetArchitecture;
+  /// Not specified (`null`) during a [dryRun].
+  Architecture? get targetArchitecture;
 
   /// The operating system being compiled for.
-  OS get targetOs;
+  OS get targetOS;
 
   /// When compiling for iOS, whether to target device or simulator.
   ///
-  /// Required when [targetOs] equals [OS.iOS].
+  /// Required when [targetOS] equals [OS.iOS].
   ///
-  /// Not available in [dryRun].s
+  /// Not available during a [dryRun]. Will throw a [StateError] if accessed
+  /// during a [dryRun].
   IOSSdk? get targetIOSSdk;
 
   /// When compiling for Android, the minimum Android SDK API version to that
   /// the compiled code will be compatible with.
   ///
-  /// Required when [targetOs] equals [OS.android].
+  /// Required when [targetOS] equals [OS.android].
   ///
-  /// Not available in [dryRun].
+  /// Not available during a [dryRun]. Will throw a [StateError] if accessed
+  /// during a [dryRun].
   ///
   /// For more information about the Android API version, refer to
   /// [`minSdkVersion`](https://developer.android.com/ndk/guides/sdk-versions#minsdkversion)
   /// in the Android documentation.
   int? get targetAndroidNdkApi;
 
-  /// Preferred linkMode method for library.
+  /// The preferred [LinkMode] method for [CCodeAsset]s.
   LinkModePreference get linkModePreference;
 
-  /// Metadata from direct dependencies.
+  /// Metadata from a direct dependency.
   ///
-  /// The key in the map is the package name of the dependency.
+  /// The [packageName] of is the package name of the direct dependency.
   ///
-  /// The key in the nested map is the key for the metadata from the dependency.
+  /// Returns `null` if metadata was not provided.
   ///
-  /// Not available in [dryRun].
-  Map<String, Metadata>? get dependencyMetadata;
+  /// Not available during a [dryRun]. Will throw a [StateError] if accessed
+  /// during a [dryRun].
+  Object? metadatum(String packageName, String key);
 
   /// The configuration for invoking the C compiler.
   ///
-  /// Not available in [dryRun].
+  /// Not available during a [dryRun]. Will throw a [StateError] if accessed
+  /// during a [dryRun].
   CCompilerConfig get cCompiler;
 
-  /// Don't run the build, only report the native assets produced.
+  /// Indicates whether a run should build (wet run) or report potential assets
```

Comment:
Drop "Indicates". Always drop "indicates". Say, or say not, there is no "indicates". :)

I don't think "wet run" is an idiom the same way as "dry run".
And this doesn't say which value means what.

So maybe:
```dart
/// Whether this run is a dry-run, which doesn't build anthing.
///
/// A dry-run only reports information about which assets a build would create,
/// but doesn't actually create files.
///